### PR TITLE
refactor: move test builders out of compactor2 code

### PR DESCRIPTION
**Pull request by alamb**
Related to https://github.com/influxdata/idpe/issues/17089

Goal: I am trying to get to the point where there is a testable public interface for the compactor(so I can move the compactor2 integration tests type code into their own `test` binary and not have the test code all intermixed in the library

Changes: 
1. Move builder builder structures out of test_utils in compactor into iox_tests to make them more discoverable and reduce dependencies
2. Start organizing the code a little more in iox_tests

### CODE REVIEW

File: compactor2/src/test_util.rs [hunk 1]
```diff
@@ -42,116 +41,7 @@ use crate::{
 };
 
 #[derive(Debug)]
-pub struct ParquetFileBuilder {
```
Comment (alamb) [inline compactor2/src/test_util.rs:45]: code is moved to `iox_tests` and has some comments added

File: iox_tests/src/lib.rs [hunk 2]
```diff
@@ -12,4 +12,11 @@
     clippy::dbg_macro
 )]
```
Comment (alamb) [inline iox_tests/src/lib.rs:14]: I renamed `util.rs` to `catalog.rs` to better reflect what it does, and moved the publicly exported structures into the main module to make it easier to know what was in this crate

File: iox_tests/src/builders.rs [hunk 3]
```diff
@@ -0,0 +1,195 @@
+use data_types::{
```
Comment (alamb) [inline iox_tests/src/builders.rs:1]: this code was moved from `test_utils.rs` in compactor2

File: compactor2/src/components/commit/logging.rs [hunk 4]
```diff
@@ -82,12 +82,9 @@ mod tests {
...
     use test_helpers::tracing::TracingCapture;
 
-    use crate::{
-        components::commit::mock::{CommitHistoryEntry, MockCommit},
-        test_util::ParquetFileBuilder,
-    };
-
     use super::*;
+    use crate::components::commit::mock::{CommitHistoryEntry, MockCommit};
+    use iox_tests::ParquetFileBuilder;
```
Comment (NGA-TRAN) [inline compactor2/src/components/commit/logging.rs:87]: 👍

**Review by NGA-TRAN — APPROVED** [review]
Since the compactor2 tests rely heavily on the test builder specific setup. The fact that the tests still pass does verify this is a refactor 😄 . Thanks @alamb

**Pull request merged**